### ISSUE TITLE
SEE LICENSE IN ... has no effect when file does not contain known license

### ISSUE THREAD

**Issue by mgroenhoff** [davglass/license-checker#125]
I have an issue where some modules that have `SEE LICENSE IN ...` in their `license` field have their licenses reported incorrectly when it does not understand the contents of the license file.

The following happens in my case:

1. First it finds the `license` field in `package.json`
1. Then it matches `/SEE LICENSE IN (.*)/i` (so it returns `Custom: $1`)
1. Then it walks over the files `['LICENSE', 'LICENCE', 'COPYING', 'README']`
1. It finds and checks the `LICENSE` file but it contains custom license text so it doesn't match any license string
1. It then finds and checks the `README` file and it matches the first url it encounters which happens to be a link to an issue tracker in the change log section in the readme file
1. This license is reported as something like `Custom: https://github.com/.../.../issues/x`

What i expect:

Use the license pointed to by the file in `SEE LICENSE IN ...` because i explicitly told it to do so and if it does not understand what is inside the file just use `UNKNOWN` but do not continue and search in other files.

**Comment by davglass** [comment]
Can I get an example module please? The custom license detection/URL parsing was a user PR and I'm not all that familiar with it.

**Comment by mgroenhoff** [comment]
I created a very minimal module named `foo`
`package.json`
```js
{
  "name": "foo",
  "version": "1.0.0",
  "license": "SEE LICENSE IN LICENSE"
}
```
`LICENSE`
```
This is a custom license
```
`README.md`
```markdown
# foo

This is a readme file

## Changelog

### 1.0.0 (2017-09-21)

* [#125](https://github.com/davglass/license-checker/issues/125): description...

```

If i then run `license-checker` i get the following output:
```
> license-checker
└─ foo@1.0.0
   ├─ licenses: Custom: https://github.com/davglass/license-checker/issues/125
   └─ licenseFile: C:\Source\foo\LICENSE
```
and `license-checker --summary` outputs:
```
> license-checker --summary
└─ Custom: https://github.com/davglass/license-checker/issues/125: 1
```

**Comment by bmuenzenmeyer** [comment]
The way I worked around this was by placing a hidden HTML comment in the top of my README that has a `SEE LICENSE IN {link}` link in it. Not perfect, but at least gets the point across.

**Comment by TjlHope** [comment]
I've had a first go at fixing this as I was bitten by it at work (thanks for the project BTW!): TjlHope/license-checker@4b9bbef1db7b0993f67fb715479c26e1860e3949
Still need to add test cases to get the coverage back up, and I'm not convinced the rest of the license-files stuff is doing the "right thing", but tried to keep as is for now to not break other people's use cases...
